modules/options/src/main/scala/scala/build/internal/ScalaNightlyVersion.scala

```diff
@@ -0,0 +1,73 @@
+package scala.build.internal
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker
+import coursier.cache.FileCache
+import coursier.util.{Artifact, Task}
+import dependency.ScalaVersion
+
+import scala.build.EitherCps.{either, value}
+import scala.build.Os
+import scala.build.errors.{BuildException, ScalaVersionError}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+object ScalaNightlyVersion {
+
+  private object Scala2Repo {
+    final case class ScalaVersion(name: String, lastModified: Long)
+    final case class ScalaVersionsMetaData(repo: String, children: List[ScalaVersion])
+
+    val codec: JsonValueCodec[ScalaVersionsMetaData] = JsonCodecMaker.make
+  }
+
+  private def downloadScala2RepoPage(cache: FileCache[Task]): Either[BuildException, Array[Byte]] =
+    either {
+      val scala2NightlyRepo =
+        "https://scala-ci.typesafe.com/ui/api/v1/ui/nativeBrowser/scala-integration/org/scala-lang/scala-compiler"
+      val artifact = Artifact(scala2NightlyRepo).withChanging(true)
+      val res = cache.logger.use {
+        try cache.withTtl(0.seconds).file(artifact).run.unsafeRun()(cache.ec)
```

Comment:
Would ttl set to 0 cause every run to access the file?

Reply:
It means, the courier try to download web page repository every time. If there is no access to internet, it uses previously downloaded page.

Reply:
Can we increase ttl to an hour or two? So noop compilation with .nightly are fast?